linode/resource_linode_token.go

```diff
@@ -0,0 +1,193 @@
+package linode
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/linode/linodego"
+)
+
+func resourceLinodeToken() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLinodeTokenCreate,
+		Read:   resourceLinodeTokenRead,
+		Update: resourceLinodeTokenUpdate,
+		Delete: resourceLinodeTokenDelete,
+		Exists: resourceLinodeTokenExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"label": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The label of the Linode Token.",
+				Optional:    true,
+			},
+			"scopes": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The scopes this token was created with. These define what parts of the Account the token can be used to access. Many command-line tools, such as the Linode CLI, require tokens with access to *. Tokens with more restrictive scopes are generally more secure.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"expiry": &schema.Schema{
+				Type:             schema.TypeString,
+				Description:      "When this token will expire. Personal Access Tokens cannot be renewed, so after this time the token will be completely unusable and a new token will need to be generated. Tokens may be created with 'null' as their expiry and will never expire unless revoked.",
+				Optional:         true,
+				ValidateFunc:     validDateTime,
+				ForceNew:         true,
+				DiffSuppressFunc: equivalentDate,
```

Comment:
Did not know about this. Good to know

Reply:
I made up [`equivalentDate`](https://github.com/terraform-providers/terraform-provider-linode/pull/5/files/198e4b4e30e628ce9720cccb77d621a22b89c6eb#diff-0d95025d49ee76643538ff7d125f1495R58) and [`validDateTime`](https://github.com/terraform-providers/terraform-provider-linode/pull/5/files/198e4b4e30e628ce9720cccb77d621a22b89c6eb#diff-0d95025d49ee76643538ff7d125f1495R70).

I assume you mean  [DiffSuppressFunc](https://www.terraform.io/docs/extend/schemas/schema-behaviors.html#diffsuppressfunc).